### PR TITLE
Several activation functions useful for optimisation.

### PR DESCRIPTION
**Pull request by jetuk**
Initial commit of 3 functions - binary step, rectifier and logistic. No tests and only basic docstrings at the moment. Commit mainly for comment.

### CODE REVIEW

**Review by matrosoe — COMMENTED** [review]
A couple comments for the rectifier function.

File: pywr/parameters/_activation_functions.pyx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        self.double_size = 1
+        self.integer_size = 0
+        self._lower_bounds = lower_bounds
+        self._upper_bounds = upper_bounds
+
+    cpdef double value(self, Timestep ts, ScenarioIndex scenario_index) except? -1:
+        if self._value <= 0.0:
+            return 0.0
+        else:
+            return self.max_output * self._upper_bounds
```
[the remark this answers is not in the document]
Comment (jetuk) [reply pywr/parameters/_activation_functions.pyx:77]: Yes, good point.

**Comment by jetuk** [comment]
@matrosoe @Batch21 this is now ready for review.

**Review by matrosoe — APPROVED** [review]
Had issues with pytest not importing modules, I just needed to update it!

**Pull request merged**